src/features/swap/SwapForm.tsx

```diff
@@ -304,40 +316,84 @@ function SubmitButton() {
       logger.error('Error updating network', error)
       toast.error('Could not switch network, does wallet support switching?')
     }
-  }
+  }, [switchNetworkAsync, dispatch])
 
-  const error =
-    touched.amount && (errors.amount || errors.fromTokenId || errors.toTokenId || errors.slippage)
-  let text
-
-  if (error) {
-    text = error
-  } else if (!isAccountReady) {
-    text = 'Connect Wallet'
-  } else if (!isOnCelo) {
-    text = 'Switch to Celo Network'
-  } else {
-    text = 'Continue'
-  }
+  const amountWasModified = touched.amount || values.amount
+  const quoteLikelyStillLoading = useMemo(
+    () =>
+      values.amount && values.quote && values.quote === '0' && errors.amount === 'Amount Required',
+    [values.amount, values.quote, errors.amount]
+  )
 
-  const type = isAccountReady ? 'submit' : 'button'
-  let onClick
+  const hasError = useMemo(() => {
+    if (!amountWasModified) return false
+    if (quoteLikelyStillLoading) return false
 
-  if (!isAccountReady) {
-    onClick = openConnectModal
-  } else if (!isOnCelo) {
-    onClick = switchToNetwork
-  }
+    return !!(
+      errors.amount ||
+      errors.quote ||
+      errors.fromTokenId ||
+      errors.toTokenId ||
+      errors.slippage
+    )
+  }, [amountWasModified, quoteLikelyStillLoading, errors])
+
+  const errorText = useMemo(
+    () =>
+      errors.amount ||
+      errors.quote ||
+      errors.fromTokenId ||
+      errors.toTokenId ||
+      errors.slippage ||
+      '',
+    [errors]
+  )
 
-  const showLongError = typeof error === 'string' && error?.length > 50
+  const showLongError = useMemo(() => errorText.length > 50, [errorText])
+
+  const buttonText = useMemo(() => {
+    if (isValidating && (!errors.amount || !quoteLikelyStillLoading)) return 'Continue'
+    if (!isAccountReady) return 'Connect Wallet'
+    if (!isOnCelo) return 'Switch to Celo Network'
+    if (hasError) {
+      return showLongError ? 'Error' : errorText
```

Comment:
It looks like incorrect because:
const showLongError = useMemo(() => errorText.length > 50, [errorText])

If showLongError there is - we show "error"? It reads as a flag to show a long error with errorText instead of "Error".